package-system/AWSGameLiftServerSDK/build_and_package.py

```diff
@@ -0,0 +1,290 @@
+#
+# All or portions of this file Copyright (c) Amazon.com, Inc. or its affiliates or
+# its licensors.
+#
+# For complete copyright and license terms please see the LICENSE at the root of this
+# distribution (the "License"). All use of this software is governed by the License,
+# or, if provided, by the license below or the license accompanying this file. Do not
+# remove or modify any license notices. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+
+import argparse
+import atexit
+import itertools
+import json
+import os
+import shutil
+import sys
+
+from contextlib import contextmanager
+
+
+def ParseArguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--version', required=True, help='The version number of GameLiftServerSDK, e.g. 3.4.1')
+    parser.add_argument('--platform', required=True, choices=['windows', 'linux'])
+    parser.add_argument('--config', required=True, help='gamelift-sdk.json')
+    parser.add_argument('--msbuild_path', help='msbuild executable path for windows platform')
+
+    return parser.parse_args()
+
+@contextmanager
+def InDirectory(path):
+    prev = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(prev)
+
+class Variant:
+    """
+    Variant class is used to store variants info for cmake build
+    """
+    def __init__(self, platform, arch, toolchain, config, libtype):
+        self.platform = platform
+        self.arch = arch
+        self.toolchain = toolchain
+        self.config = config
+        self.libtype = libtype
+
+    def __str__(self):
+        return f'{self.arch}-{self.toolchain}-{self.config}-{self.libtype}'
+
+    __repr__ = __str__
+
+    def GetPath(self):
+        return str(self)
+
+    def GetGenerateProjectsCommand(self):
+        return f'cmake {self.platform.GetCMakeDefines(self)} {self.platform.GetCMakeGenerator(self)} ..{os.sep}..'
+
+    def GenerateProjects(self):
+        projectPath = self.GetPath()
+        os.makedirs(projectPath, exist_ok=True)
+
+        with InDirectory(projectPath):
+            return os.system(self.GetGenerateProjectsCommand())
+
+        # shouldn't get here, but there is a chance the chdir could fail
```

Comment:
Nit: May worth printing out an error message here?

Reply:
just copied from perforce, this script will be modified later once we migrate to build script